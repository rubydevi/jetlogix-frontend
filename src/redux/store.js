import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservation: reservationsReducer,
  },
});

export default store;
