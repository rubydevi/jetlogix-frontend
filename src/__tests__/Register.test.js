import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

import Register from '../components/Register';

test('renders Register component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
