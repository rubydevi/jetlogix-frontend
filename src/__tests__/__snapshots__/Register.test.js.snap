// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Register component correctly 1`] = `
<div>
  <section
    class="rounded-1 login-register-style"
  >
    <p
      class="container offScreen"
    />
    <h1>
      Register
    </h1>
    <form>
      <label
        for="username"
      >
        Username:
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check hide"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark hide"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </label>
      <input
        aria-describedby="uidnote"
        aria-invalid="true"
        autocomplete="off"
        id="username"
        required=""
        type="text"
        value=""
      />
      <p
        class="offscreen"
        id="uidnote"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-info "
          data-icon="circle-info"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
        4 to 24 characters.
        <br />
        Must begin with a letter.
        <br />
        Letters, numbers, underscores, hyphens allowed.
      </p>
      <label
        for="email"
      >
        Email:
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check hide"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark hide"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </label>
      <input
        aria-describedby="emdnote"
        aria-invalid="true"
        autocomplete="off"
        id="email"
        required=""
        type="email"
        value=""
      />
      <p
        class="offscreen"
        id="emdnote"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-info "
          data-icon="circle-info"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
        Must have proper email address
        <br />
        example@gmail.com
      </p>
      <label
        for="password"
      >
        Password:
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check hide"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark hide"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </label>
      <input
        aria-describedby="pwdnote"
        aria-invalid="true"
        autocomplete="off"
        id="password"
        required=""
        type="password"
        value=""
      />
      <p
        class="offscreen"
        id="pwdnote"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-info "
          data-icon="circle-info"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
        8 to 24 characters.
        <br />
        Must include uppercase and lowercase letters, a number and a special character.
        <br />
        Allowed special characters:
        <span
          aria-label="exclamation mark"
        >
          !
        </span>
        <span
          aria-label="at symbol"
        >
          @
        </span>
        <span
          aria-label="hashtag"
        >
          #
        </span>
        <span
          aria-label="dollar sign"
        >
          $
        </span>
        <span
          aria-label="percent"
        >
          %
        </span>
      </p>
      <label
        for="matchPassword"
      >
        Confirm Password:
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check hide"
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark hide"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </label>
      <input
        aria-describedby="confirmnote"
        aria-invalid="false"
        autocomplete="off"
        id="matchPassword"
        required=""
        type="password"
        value=""
      />
      <p
        class="offscreen"
        id="confirmnote"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-info "
          data-icon="circle-info"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
        Must match the first password input field.
      </p>
      <button
        class="btn-color text-light"
        disabled=""
        type="submit"
      >
        Sign Up
      </button>
    </form>
    <p>
      Already registered?
      <br />
      <a
        class="text-light"
        href="/login"
      >
        <span
          class="text-login"
        >
           Sign In
        </span>
      </a>
    </p>
  </section>
</div>
`;
